onnxruntime/core/session/inference_session.cc

```diff
@@ -1024,16 +1024,21 @@ Status InferenceSession::LoadOrtModelWithLoader(std::function<Status()> load_ort
   const auto* fbs_ort_model_version = fbs_session->ort_version();
   ORT_RETURN_IF(fbs_ort_model_version == nullptr, "Serialized version info is null. Invalid ORT format model.");
 
+  // if this is a full build we can update an old format model.
+  bool model_needs_update = IsOrtModelVersionSupported(fbs_ort_model_version->string_view()) == false;
```

Comment:
do we need to distinguish between older and newer unsupported versions? i.e., updating is possible for older ones but I guess we should consider newer ones unsupported.

also, I guess technically the older versions are not "unsupported" anymore (in a full build).

Reply:
Yes it should.